include/chainbase/pinnable_mapped_file.hpp

```diff
@@ -88,10 +108,13 @@ class pinnable_mapped_file {
       bip::permissions                              _db_permissions{S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH};
 #endif
 
-      segment_manager*                              _segment_manager = nullptr;
+      std::unique_ptr<segment_manager, decltype([](auto) noexcept {})> _segment_manager;
```

Comment:
With gcc13 this causes a massive number of warnings,
```
pinnable_mapped_file.hpp:50:7: warning: ‘chainbase::pinnable_mapped_file’ has a field ‘std::unique_ptr<boost::interprocess::segment_manager<char, boost::interprocess::rbtree_best_fit<boost::interprocess::mutex_family>, boost::interprocess::iset_index>, chainbase::pinnable_mapped_file::<lambda(auto:49)> > chainbase::pinnable_mapped_file::_segment_manager’ whose type has internal linkage [-Wsubobject-linkage]
   50 | class pinnable_mapped_file {
```

Reply:
OK, that's what I get for trying to be smart and enforcing that the `_segment_manager` pointer was only moved (not copied) in the move constructors and assignment operators.

I reverted to using a raw pointer.